src/app/coursegrab/utils/scraper.py

```diff
@@ -0,0 +1,71 @@
+import requests
+import bs4
+
+ROOT_URL = "https://classes.cornell.edu"
+
+
+def current_semester():
+    req = requests.get(ROOT_URL)
+    req.raise_for_status()
+    return req.url.strip("/").split("/")[-1]
+
+
+def all_subject_codes():
+    req = requests.get(ROOT_URL)
+    roster_bs4 = bs4.BeautifulSoup(req.text, "html.parser")
+    subject_tags = roster_bs4.select(".browse-subjectcode")
+    return [str(tag.getText()) for tag in subject_tags]
+
+
+def get_course_status(subject_code, catalog_num):
+    semester = current_semester()
+    subject_url = "http://classes.cornell.edu/browse/roster/" + semester + "/subject/" + subject_code
```

Comment:
should the `http://classes..` be taken out into a constant

Reply:
yea we should probably make a constants file soon

Reply:
`ROOT_URL` at top